docs/content/4.sdk/3.advanced/3.file-upload.md

```diff
@@ -0,0 +1,233 @@
+  {
+    files: [file1, file2, file3],
+    batchId: "batch123",
+    processImmediately: true,
+  },
+  prepareConfig({
+    headers: {
+      "Content-Type": "multipart/form-data",
+    },
+  })
+);
+```
+
+::tip
+All additional parameters will be automatically included in the multipart/form-data request. You can include any serializable data alongside your files.
+::
+
+## Framework-Specific Examples
+
+::tabs{:titles='["Next.js", "Nuxt.js"]'}
+#tab-1
+
+```typescript
+// pages/api/upload.ts
+import { NextApiRequest, NextApiResponse } from "next";
+import { prepareConfig } from "@vue-storefront/sdk";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    return res.status(405).end();
+  }
+
+  try {
+    const sdk = await getSdk();
+    const formData = new FormData();
+
+    if (req.files) {
+      req.files.forEach((file: File) => {
+        formData.append("files", file);
+      });
+    }
+
+    const response = await sdk.commerce.uploadFile(
+      { files: formData },
+      prepareConfig({
+        method: "POST",
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+      })
+    );
+
+    res.status(200).json(response);
+  } catch (error) {
+    console.error("Upload failed:", error);
+    res.status(500).json({ error: "Upload failed" });
+  }
+}
+```
+
+#tab-2
+
+```typescript
+// pages/upload.vue
+<template>
+  <div>
+    <h1>Upload Files</h1>
+    <input type="file" multiple @change="handleFileChange" />
+    <button @click="uploadFiles">Upload</button>
+    <p v-if="uploadStatus">{{ uploadStatus }}</p>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+import { prepareConfig } from '@vue-storefront/sdk';
+
+const selectedFiles = ref<File[]>([]);
+const uploadStatus = ref<string | null>(null);
+
+const handleFileChange = (event: Event) => {
+  const target = event.target as HTMLInputElement;
+  if (target.files) {
+    selectedFiles.value = Array.from(target.files);
+  }
+};
+
+const uploadFiles = async () => {
+  if (!selectedFiles.value.length) return;
+
+  try {
+    const sdk = await useSdk();
+    const formData = new FormData();
+
+    selectedFiles.value.forEach((file) => {
+      formData.append('files', file);
+    });
+
+    const response = await sdk.commerce.uploadFile(
+      { files: formData },
+      prepareConfig({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        }
+      })
+    );
+
+    uploadStatus.value = 'Files uploaded successfully!';
+  } catch (error) {
+    console.error('Upload failed:', error);
+    uploadStatus.value = 'An error occurred during upload.';
+  }
+};
+</script>
+```
+
+::
+
+## Usage with Middleware
+
+::success
+To read more about how to use file upload with the Middleware, please refer to the [File Upload Middleware Guide](https://docs.alokai.com/middleware/guides/file-upload).
+::
```

Comment:
If we want to split the guides into two places (middleware and sdk pages), I would move this reference to the top of the docs, as the middleware part has to be implemented first. But imo it would be better to have everything described in one place.. (but then - which one? - idk)

Reply:
I also do not know where because we do not have such place. I will move middleware ref to the top.